### DISCUSSION TITLE
error: failed to compile `shadowsocks-rust v1.14.2`

### DISCUSSION THREAD

**Discussion by iugo** [shadowsocks/shadowsocks-rust#795] [General]
error: failed to compile `shadowsocks-rust v1.14.2`

```
error: linking with `cc` failed: exit status: 1
error: could not compile `libc` due to previous error

  = note: /usr/bin/ld: cannot find Scrt1.o: No such file or directory
          /usr/bin/ld: cannot find crti.o: No such file or directory
          collect2: error: ld returned 1 exit status
```

可能是当前 Linux 发行版缺少依赖.

根据关键字搜索, 在多处 rust 编译时均会出现. 一般只要安装依赖即可, 不过具体缺哪个好像不清楚.

我是在 `sudo apt install libsqlite3-dev` 之后即可正常编译. 但和这个包本身应该没有关系. 可能是 `libc6-dev linux-libc-dev` 这两个包起到了作用.

**Comment by zonyitoo** [comment]
It all depending on what toolchain you were using. If you are using `*-unknown-linux-gnu`, then glibc, c compilers, ld are mandatory.
